freetextresponse/mixins.py

```diff
@@ -0,0 +1,39 @@
+"""
+Due Date Integration xBlock Mixin
+"""
+import datetime
+import pytz
+from django.conf import settings
+from xblock.fields import DateTime
+from xblock.fields import Scope
+
+
+class DueDateXblockMixin(object):
+    """
+    xBlock Mixin to allow xblocks to check the due date of
+    the subsection in which they are placed
+    """
+    due = DateTime(
+        scope=Scope.settings,
+        default=None,
+    )
+    graceperiod = None
+
+    def past_due(self):
+        """
+        Return whether due date has passed.
+        """
+        site_tz = pytz.timezone(getattr(settings, 'TIME_ZONE', pytz.utc.zone))
+        now = datetime.datetime.utcnow().replace(tzinfo=site_tz)
+        if self.due is not None:
+            due_date = self.due
+            if self.graceperiod is not None:
+                due_date = due_date + self.graceperiod
+            return now > due_date
+        return False
+
+    def before_due(self):
```

Comment:
I'm not a fan of this composite value.

"past due" is an expected description of an assignment,
not just in code but also IRL.
It's unlikely that anyone would describe an assignment as "before due".

Also, it's one more function to remember/test
and worse, one more thing that can go wrong.

As seen here:
```python
past_due = now > due_date
before_due = not past_due
# which can be rewritten as
before_due = not (now > due_date)
before_due = now <= due_date
# which means that
before_due is True  # when due_date == now
# which is misleading
```

It becomes easier to miss these edge cases, the more composite values are used.

We should remove this method and just say:
` if not past_due`, wherever necessary.